Packages/Tracking/Core/Runtime/Scripts/HandPoses/PoseDetectionEditor.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using System;
+
+namespace Leap.Unity
+{
+    [ExecuteInEditMode]
+    public class PoseDetectionEditor : LeapProvider
+    {
+        [HideInInspector]
+        public override Frame CurrentFrame
+        { get
+            { 
+                return new Frame(0, (long)Time.realtimeSinceStartup, 90, currentHands);
+            }
+        }
+        [HideInInspector]
+        public Frame TestFrame = null;
+        [HideInInspector]
+        public override Frame CurrentFixedFrame => new Frame();
+
+        // Pose to apply to the left hand
+        [SerializeField]
+        HandPoseScriptableObject leftHandPoseObject;
+        // Pose to apply to the right hand
+        [SerializeField]
+        HandPoseScriptableObject rightHandPoseObject;
+
+        private List<Hand> currentHands = new List<Hand>();
+
+        private List<Color> gizmoColours = new List<Color>() { Color.green, Color.yellow, Color.red, Color.blue };
+
+        // Update is called once per frame
+        private void OnValidate()
+        {
+            Hand PosedHand;
+            currentHands.Clear();
+
+            if (leftHandPoseObject != null)
+            {
+                PosedHand = leftHandPoseObject.GetSerializedHand();
+                PosedHand.SetTransform(new Vector3(0, 0, 0), PosedHand.Rotation);
+                leftHand = PosedHand;
+                currentHands.Add(PosedHand);
+            }
+            if(rightHandPoseObject != null)
+            {
+                PosedHand = rightHandPoseObject.GetSerializedHand();
+                PosedHand.SetTransform(new Vector3(0.5f, 0, 0), PosedHand.Rotation);
+                rightHand = PosedHand;
+                currentHands.Add(PosedHand);
+            }
+
+            TestFrame = CurrentFrame;
+            DispatchUpdateFrameEvent(CurrentFrame);
+        }
+
+        Hand rightHand, leftHand;
+
+        private void OnDrawGizmos()
+        {
+            ShowEditorGizmos(rightHand, rightHandPoseObject);
+            ShowEditorGizmos(leftHand, leftHandPoseObject);
+            
+        }
+
+        private void ShowEditorGizmos(Hand hand, HandPoseScriptableObject handPoseScriptableObject)
+        {
+            float lineThickness = 4;
+
```

Comment:
We should add a check for if `hand `or `handPoseScriptableObject `are null here, so you don't need to reference both in editor to see gizmos